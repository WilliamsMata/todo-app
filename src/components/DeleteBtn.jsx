import PropTypes from "prop-types";

export const DeleteBtn = ({ onClick }) => {
  return (
    <button className="btn btn-danger d-flex" onClick={onClick}>
      <img
        style={{ width: 20, aspectRatio: 1 / 1.5 }}
        className="align-self-center justify-self-center"
        src="./delete-alt-svgrepo-com.svg"
        alt="delete"
      />
    </button>
  );
};

DeleteBtn.propTypes = {
  onClick: PropTypes.func,
};
