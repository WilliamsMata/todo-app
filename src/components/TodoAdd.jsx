import { useForm } from "../hooks/useForm";
import PropTypes from "prop-types";

export const TodoAdd = ({ onNewTodo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: "",
  });

  const onSubmit = (event) => {
    event.preventDefault();
    if (description.trim().length <= 1) return;
    const newTodo = {
      id: new Date().getTime(),
      description: description.trim(),
      done: false,
    };
    onNewTodo(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        className="form-control"
        type="text"
        placeholder="Que hay que hacer?"
        autoComplete="off"
        name="description"
        value={description}
        onChange={onInputChange}
      />

      <button type="submit" className="btn btn-primary mt-2">
        Agregar Tarea
      </button>
    </form>
  );
};

TodoAdd.protoTypes = {
  onNewTodo: PropTypes.func,
};
