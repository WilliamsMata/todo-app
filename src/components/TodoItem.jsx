import { DeleteBtn } from "./DeleteBtn";

export const TodoItem = ({ todo, onDeleteTodo, onToggleTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between align-items-center p-0}">
      <span
        className={`align-self-center flex-grow-1 pt-2 pb-2 ${
          todo.done ? "text-decoration-line-through" : ""
        }`}
        style={{ cursor: "pointer" }}
        onClick={() => onToggleTodo(todo.id)}
        aria-label="span"
      >
        {todo.description}
      </span>

      <DeleteBtn onClick={() => onDeleteTodo(todo.id)} />
    </li>
  );
};
