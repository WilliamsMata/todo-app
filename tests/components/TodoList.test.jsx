import { render, screen } from "@testing-library/react";
import { TodoList } from "../../src/components/TodoList";

describe("Pruebas en <TodoList />", () => {
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<TodoList />);

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar las tareas creadas", () => {
    const todos = [
      {
        id: 1,
        description: "Tarea 1",
        done: false,
      },
      {
        id: 2,
        description: "Tarea 2",
        done: false,
      },
    ];

    render(<TodoList todos={todos} />);

    expect(screen.getByText("Tarea 1")).toBeTruthy();
    expect(screen.getByText("Tarea 2")).toBeTruthy();
  });
});
