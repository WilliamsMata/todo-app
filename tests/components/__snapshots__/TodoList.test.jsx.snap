// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Debe de hacer match con el snapshot 1`] = `
<div>
  <ul
    class="list-group"
  />
</div>
`;
